includes/logs/log/functions.php

```diff
@@ -63,6 +63,27 @@ function edd_add_log( $data = array() ) {
 		}
 	}
 
+	/**
+	 * Allow the ability to disable a generic log entry by object_type.
+	 *
+	 * Based on the object type, allows developers to disable the insertion of a log.
+	 *
+	 * Example:
+	 * add_filter( 'edd_should_log_gateway_error', '__return_false' )
+	 *
+	 * You can find a list of the logs object types in the edd_logs table.
+	 *
+	 * @since 3.1
+	 *
+	 * @param bool  $should_record_log If this log should be inserted
+	 * @param array $data              The data to be logged.
+	 */
+	$should_record_log = apply_filters( "edd_should_log_{$data['object_type']}", true, $data );
```

Comment:
@robincornett seems to me like the above logic is just overly complicated...unless there is a specific use case for all of it.

Seems to me we need to adjust `object_type` to be required or set it to a default value if we're not returning `false` earlier.

Reply:
@robincornett Correct me if I'm wrong but the issue here is that somehow it is possible to add a log with the following condition to have it fail this check:
```
	// An object type must be supplied for every log that is
	// inserted into the database.
	if ( empty( $data['object_type'] ) ) {
		// If the object_type is set to null, check the value of the object_id.
		if ( is_null( $data['object_type'] ) ) {
			// If the object_id is not empty but the object_type is, fail this log attempt.
			if ( ! empty( $data['object_id'] ) ) {
				return false;
			}
			// The object_type is not null, and it is empty. No log will take place.
		} else {
			return false;
		}
	}
```
`object_type` = `null`
`object_id` = (any empty value)

In this case you'd end up with no return early...but you'd have a null object type and an empty object Id, however at the beginning of this whole things we 'require' an `object_type` per the comments.

So, my thought here is....we' need to fix this to actually require an `objec_type` at a minimum, no matter what the `object_id` is.

Reply:
You're more familiar with the history than I am--my thought was that the logic was intentionally set up to allow adding a log without a specific object type...I think allowing an object ID without an object type seems problematic because the object could be an order, customer, post, who knows? But if it's possible to allow a generic log which is not linked to an object of some kind then it seems like it should be okay to allow an empty type.

Currently I'm able to add a log entry with only the `content` defined.

Reply:
okay @robincornett see if that takes care of things, it should fix the existing issue in 3.0 as well.